product/src/test/java/org/itsallcode/openfasttrace/cli/ProcessOutputConsumer.java

```diff
@@ -0,0 +1,147 @@
+
+class ProcessOutputConsumer
+{
+    private static final Logger LOG = Logger.getLogger(ProcessOutputConsumer.class.getName());
+    private final Executor executor;
+    private final Process process;
+    private final ProcessStreamConsumer stdOutConsumer;
+    private final ProcessStreamConsumer stdErrConsumer;
+
+    ProcessOutputConsumer(final Process process)
+    {
+        this(createThreadExecutor(), process, new ProcessStreamConsumer("stdout"), new ProcessStreamConsumer("stderr"));
+    }
+
+    ProcessOutputConsumer(final Executor executor, final Process process,
+            final ProcessStreamConsumer stdOutConsumer, final ProcessStreamConsumer stdErrConsumer)
+    {
+        this.executor = executor;
+        this.process = process;
+        this.stdOutConsumer = stdOutConsumer;
+        this.stdErrConsumer = stdErrConsumer;
+    }
+
+    private static Executor createThreadExecutor()
+    {
+        return runnable -> {
+            final Thread thread = new Thread(runnable);
+            thread.setName(ProcessOutputConsumer.class.getSimpleName());
+            thread.setUncaughtExceptionHandler(new LoggingExceptionHandler());
+            thread.start();
+        };
+    }
+
+    void start()
+    {
+        executor.execute(() -> {
+            readStream(process.getInputStream(), stdOutConsumer);
+        });
+        executor.execute(() -> {
+            readStream(process.getErrorStream(), stdErrConsumer);
+        });
+    }
+
+    private void readStream(final InputStream stream, final ProcessStreamConsumer consumer)
+    {
+        try (final BufferedReader reader = new BufferedReader(new InputStreamReader(stream, StandardCharsets.UTF_8)))
+        {
+            String line = null;
+            while ((line = reader.readLine()) != null)
+            {
+                consumer.accept(line);
+            }
+            consumer.streamFinished();
+        }
+        catch (final IOException exception)
+        {
+            LOG.log(Level.WARNING, "Reading input stream failed: %s".formatted(exception.getMessage()), exception);
+            consumer.streamFinished();
+        }
+    }
+
+    String getStdOut()
+    {
+        return stdOutConsumer.getContent();
+    }
+
+    String getStdErr()
+    {
+        return stdErrConsumer.getContent();
+    }
+
+    void waitForStreamsClosed(final Duration timeout)
+    {
+        stdOutConsumer.waitUntilStreamClosed(timeout);
+        stdErrConsumer.waitUntilStreamClosed(timeout);
+    }
+
+    private static class ProcessStreamConsumer
+    {
+        private final CountDownLatch streamFinished = new CountDownLatch(1);
+        private final StringBuilder builder = new StringBuilder();
+        private final String name;
+
+        ProcessStreamConsumer(final String name)
+        {
+            this.name = name;
+        }
+
+        String getContent()
+        {
+            return builder.toString();
+        }
+
+        void streamFinished()
```

Comment:
Naming an attribute and a method the same is a recipe for confusion.
Please rename the method. Use an imperative. :-)